install-opentelemetry-dotnet.ps1

```diff
@@ -0,0 +1,163 @@
+﻿#
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+<#
+ .SYNOPSIS
+    Installs OpenTelemetry .NET Automatic Instrumentation 
+ .PARAMETER OS
+    Default: windows
+    Selects distribution based on operating system. 
+    Possible values: windows, linux, macos
+ .PARAMETER Architecture
+    Default: <auto> - this value represents currently running OS architecture
+    Architecture of CLR profiler binaries to be installed.
+    Possible values: <auto>, x64, x86
+#>
+[cmdletbinding()]
+param(
+    [string]$OS="windows",
+    [string]$Architecture="<auto>"
+)
+
+function Get-Machine-Architecture() {
+    # On PS x86, PROCESSOR_ARCHITECTURE reports x86 even on x64 systems.
+    # To get the correct architecture, we need to use PROCESSOR_ARCHITEW6432.
+    # PS x64 doesn't define this, so we fall back to PROCESSOR_ARCHITECTURE.
+    # Possible values: amd64, x64, x86, arm64, arm
+    if( $ENV:PROCESSOR_ARCHITEW6432 -ne $null ) {
+        return $ENV:PROCESSOR_ARCHITEW6432
+    }
+
+    try {        
+        if( ((Get-CimInstance -ClassName CIM_OperatingSystem).OSArchitecture) -like "ARM*") {
+            if( [Environment]::Is64BitOperatingSystem )
+            {
+                return "arm64"
+            }  
+            return "arm"
+        }
+    }
+    catch {
+        # Machine doesn't support Get-CimInstance
+    }
+
+    return $ENV:PROCESSOR_ARCHITECTURE
+}
+
+function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
+    if ($Architecture -eq "<auto>") {
+        $Architecture = Get-Machine-Architecture
+    }
+
+    switch ($Architecture.ToLowerInvariant()) {
+        { ($_ -eq "amd64") -or ($_ -eq "x64") } { return "x64" }
+        { $_ -eq "x86" } { return "x86" }
+        default { throw "Architecture '$Architecture' not supported." }
+    }
+}
+
+function Get-Install-Directory([string]$OS, [string]$Architecture) {
+    $dir = "OpenTelemetry .NET AutoInstrumentation"
+
+    if($OS -eq "windows"){
+        if($Architecture -eq "x86"){
+            return (Join-Path "C:\Program Files (x86)\" $dir)
+        } elseif($Architecture -eq "x64") {
+            return (Join-Path "C:\Program Files\" $dir)
+        }
+    }
+
+    throw "TODO: Not supported yet."
+}
```

Comment:
I suggest that the script has the same logic and functionality as https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/download.sh

E.g. I do not think it is a good idea to default to `C:\Program Files\`. This is not an MSI 😉

Reply:
I was planning to add custom install location support, but still `<auto>` should default to `C:\Program Files` as it's the most expected path for Windows. (Still have to research the most expected path for Linux and Mac also).

Reply:
I do not agree with 

> C:\Program Files as it's the most expected path for Windows

A non-admin user would not be able to use it. For instance "{Username}\AppData\Local\Programs" is also popular. I would prefer to have a simple default like in `download.sh` 😉

Reply:
Fine with this one also `"{Username}\AppData\Local\Programs"`

Reply:
So not using the hard coded paths above. correct? Check if there is an env var to help with the user location too.

Reply:
@pjanotti I would still default to some well known hard coded path but give option to choose the location. Otherwise the location is going to bloat instrumentation script and you need to set manually almost 10 path scripts. Manually setting almsot 10 path envs sounds already very bad UX.